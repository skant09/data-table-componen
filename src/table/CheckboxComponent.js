import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CheckboxComponent = props => {
  if(!props){
    props = {
      checked: false
    }
  }
  const [checkedState, setcheckedState] = useState(props.checked || false);
  const toggelCheckboxState = e => {
    setcheckedState(!checkedState);
  }
  return (
    <input type="checkbox" value={props.id} checked={Boolean(props.checked)} onChange={toggelCheckboxState} />
  )
}
export default CheckboxComponent

CheckboxComponent.propTypes = {
  id: PropTypes.number,
  checked: PropTypes.bool,
};