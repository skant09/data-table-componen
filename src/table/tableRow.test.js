import React from 'react';
import {shallow} from 'enzyme';
import CheckboxComponent from './CheckboxComponent';
import renderer from 'react-test-renderer';
import {columnsConfig} from './index'
import TableRow from './tableRow'

const selectRows = jest.fn(selection => event => {
  console.log("Mocked function");
});
const rowIndex = 2;
const row = {
  id: '1',
  albumId: '1'
}

it('renders without crashing', () => {
  shallow(<TableRow key={rowIndex} rowIndex={rowIndex}  row={row} {...{columns: columnsConfig, selectRows}} />);
});

it('matches snapshot', () => {
  const app = renderer.create(<TableRow key={rowIndex} rowIndex={rowIndex}  row={row} {...{columns: columnsConfig, selectRows}} />).toJSON();
  expect(app).toMatchSnapshot();
});
