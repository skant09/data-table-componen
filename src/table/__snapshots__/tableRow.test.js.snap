// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<tr
  onClick={[Function]}
>
  <td
    onClick={[Function]}
    style={
      Object {
        "textAlign": undefined,
      }
    }
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value="1"
    />
  </td>
  <td
    onClick={[Function]}
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    <span>
      1
    </span>
  </td>
  <td
    onClick={[Function]}
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    <span>
      1
    </span>
  </td>
  <td
    onClick={[Function]}
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <span />
  </td>
  <td
    onClick={[Function]}
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <span />
  </td>
</tr>
`;
